File: go/iq/w-comms~set~event.go
// BuildInAppCommsEvent builds <iq type="set" xmlns="w:comms">. Replies with one of: Success, Error.
func BuildInAppCommsEvent(promotionId string, type string, timestampSec int, logdata string) waBinary.Node {
	return waBinary.Node{
		Tag: "iq",
		Attrs: waBinary.Attrs{"xmlns": "w:comms", "to": "s.whatsapp.net", "type": "set"},
		Content: []waBinary.Node{
			{Tag: "event", Attrs: waBinary.Attrs{"promotion_id": promotionId, "type": type, "timestamp_sec": timestampSec, "logdata": logdata}},
		},
	}
}

File: rust/iq/w-comms~set~event.rs
/// <iq type="set" xmlns="w:comms">. Replies with one of: Success, Error.
pub fn build_in_app_comms_event(promotionId: String, type: String, timestampSec: i64, logdata: String) -> Node {
    NodeBuilder::new("iq")
        .attr("xmlns", "w:comms")
        .attr("to", "s.whatsapp.net")
        .attr("type", "set")
        .child(NodeBuilder::new("event").attr("promotion_id", promotionId).attr("type", type).attr("timestamp_sec", timestampSec).attr("logdata", logdata).build())
        .build()
}